### PR TITLE
Build images for mainnet full nodes

### PR DESCRIPTION
**Pull request by roy-dydx**
### Changelist
Build these images here instead of separate repo.

### Test Plan
Cherry-picked this PR onto release branch and ran workflows to test images.

### Author/Reviewer Checklist
- [ ] If this PR has changes that result in a different app state given the same prior state and transaction list, manually add the `state-breaking` label.
- [ ] If the PR has breaking postgres changes to the indexer add the `indexer-postgres-breaking` label.
- [ ] If this PR isn't state-breaking but has changes that modify behavior in `PrepareProposal` or `ProcessProposal`, manually add the label `proposal-breaking`.
- [ ] If this PR is one of many that implement a specific feature, manually label them all `feature:[feature-name]`.
- [ ] If you wish to for mergify-bot to automatically create a PR to backport your change to a release branch, manually add the label `backport/[branch-name]`.
- [ ] Manually add any of the following labels: `refactor`, `chore`, `bug`.

## Summary by CodeRabbit

- **New Features**
	- Introduced a GitHub Actions workflow for automated Docker image building and deployment to AWS ECR.
	- Added scripts for setting up and managing full nodes, including `mainnet.sh`, `snapshot.sh`, `start.sh`, and `vars.sh`, enhancing operational efficiency for the dYdX mainnet.
- **Bug Fixes**
	- Modified `.gitignore` to now track the `bin` directory, allowing for better management of compiled binaries.

### CODE REVIEW

**Review by vincentwschau — APPROVED** [review]

File: protocol/testing/mainnet/mainnet.sh [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+# This file initializes a non-validating full node for mainnet
+
+source "./vars.sh"
+
+CHAIN_ID="dydx-mainnet-1"
+
+# Define the mapping from version to URL
+declare -A version_to_url
+# version_to_url["v5.1.0"]="https://github.com/dydxprotocol/v4-chain/releases/download/protocol%2Fv5.1.0-dev4/dydxprotocold-v5.1.0-dev4-linux-amd64.tar.gz"
```
Comment (vincentwschau) [inline protocol/testing/mainnet/mainnet.sh:12]: Doesn't this still need to be defined?

Comment (roy-dydx) [reply protocol/testing/mainnet/mainnet.sh:12]: No, since mainnet is on v5.1.x we are using the current commit for v5.1.0 in this PR

**Comment by roy-dydx** [comment]
@Mergifyio backport release/protocol/v5.1.x

**Comment by roy-dydx** [comment]
https://github.com/Mergifyio backport release/protocol/v5.2.x

**Comment by roy-dydx** [comment]
https://github.com/Mergifyio backport release/protocol/v6.x

**Pull request merged**